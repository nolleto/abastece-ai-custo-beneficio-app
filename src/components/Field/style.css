.field {
  display: flex;
  flex-direction: column;
  width: 90%;
  margin: 5px;
  font-size: 20px;
}

.field__label {
  text-align: start;
}

.field__description {
  margin-bottom: 5px;
  text-align: start;
  font-size: 12px;
}

.field__input {
  border-radius: 3px;
  font-size: 18px;
}

@media (min-width: 768px) {
  .field {
    width: 60%;
  }
}

@media (min-width: 992px) {
  .field {
    width: 50%;
  }
}

@media (min-width: 1200px) {
  .field {
    width: 40%;
  }
}
