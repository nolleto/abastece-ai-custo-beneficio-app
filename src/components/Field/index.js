import React from 'react'
import './style.css';

const Field = ({ label, description, placeholder, value, setter }) => (
  <label className='field'>
    <span className='field__label'>
      {label}:
    </span>
    <span className='field__description'>
      {description}
    </span>

    <input
      className='field__input'
      type='number'
      step='.01'
      value={value}
      placeholder={placeholder}
      onChange={({ target }) => setter(target.value)}
    />
  </label>
)

export default Field
