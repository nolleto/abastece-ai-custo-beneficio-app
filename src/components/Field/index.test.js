import React from 'react';
import Field from './index';
import { shallow } from 'enzyme';

describe('Field component', () => {
  it('matches snapshot', () => {
    const snapshot = shallow(
        <Field
          label='label'
          description='description'
          value='value'
          setter={jest.fn()}
        />
      );
    expect(snapshot).toMatchSnapshot();
  });

  describe('Event Listeners', () => {
    it('on change value trigger setter function', () => {
      const setter = jest.fn();
      const newValue = 1.23;
      const component = shallow(
        <Field
          label='label'
          description='description'
          value='value'
          setter={setter}
        />
      );
      const input = component.find('input');
      input.simulate('change', { target: { value: newValue }});

      expect(setter).toBeCalledWith(newValue);
    });
  });
});
