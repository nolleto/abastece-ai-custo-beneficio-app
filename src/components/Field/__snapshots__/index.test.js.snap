// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component matches snapshot 1`] = `
<label
  className="field"
>
  <span
    className="field__label"
  >
    label
    :
  </span>
  <span
    className="field__description"
  >
    description
  </span>
  <input
    className="field__input"
    onChange={[Function]}
    step=".01"
    type="number"
    value="value"
  />
</label>
`;
